1. Basic Web Development/server/src/Middleware/Auth.ts

```diff
@@ -0,0 +1,40 @@
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import asyncHandler from './Async';
+import ErrorResponse from '../Util/ErrorResponse';
+import User from '../Model/User';
+
+export const protect = asyncHandler(async (req: any, res: any, next: any) => {
+    let token;
+
+    if (req.headers.authorization && req.headers.authorization.startsWith('Bearer')) {
+        token = req.headers.authorization.split(' ')[1];
+    }
+
+    if(!token) {
+        return next(new ErrorResponse('Not authorized to access this route', 401));
+    }
+
+    try{
+        const decodedToken = jwt.verify(token, "agenwmgt43809tg340jgr4ogjv0");
+
+        console.log(decodedToken);
```

Comment:
```suggestion
```

Reply:
This was removed [here](https://github.com/dochiasebastian/workshops/pull/10/files#diff-c5ff3abc299dfbdbdda89f4cace74da96b218129bcf7dea849cf61f5154698a4R13-R19)